File: NTT/Models/Dto/ImageDto.cs
using System;
using ClearAnimal.Domain;
using Microsoft.CodeAnalysis;
using NTT.Domain;

namespace NTT.API.Models.Dto
{
    public class ImageDto 
    {

        public string Id { get; set; }
        public string Name { get; set; }
        public Age Age { get; set; }
        public string UploadDateString { get; set; }
        public bool Type { get; set; }
        public string Location { get; set; }
        public UserDto User { get; set; }
        public string Url { get; set; }
        public ImageDto FromDomain(Image image)
        {
            return new ImageDto
            {
                Id = image.Id,
                Name = image.Name,
                UploadDateString = image.UploadDate.ToString("yyyyMMdd"),
                Type =image.Type,
                Location=image.Location,
                User = new UserDto().FromDomain(image.User),
                Age = image.Age,
                Url = image.Url
            };
        }

        public Image ToDomain()
        {
            return new Image
            {
                Id = Id,
                Name = Name,
                Type = Type,
                UploadDate = UploadDateString == null ? new DateTime(1970, 1, 1) : DateTime.ParseExact(UploadDateString, "yyyyMMdd", System.Globalization.CultureInfo.InvariantCulture),
                Location = Location,
                User = User.ToDomain(),
            };
        }
    }
}

File: NTT.Frontend2/src/app/dto/imagedto.ts
import { AgeModel } from "../models/age.model";
import { ImageModel as ImageModel } from "../models/imageModel";
import { UserModel } from "../models/userModel";
import { UserDto } from "./Userdto";

export class ImageDto  {
  id: string;
  name: string;
  age : AgeModel;
  uploadDateString: string;
  type: boolean;
  location:string;
  user: UserDto;
  url: string;

    ToModel(): ImageModel{
        const image = new ImageModel(
        );
        image.id = this.id;
        image.age=this.age;
        image.name=this.name;
        image.uploadDate = this.uploadDateString ? new Date(parseInt(this.uploadDateString.substr(0,4)),parseInt(this.uploadDateString.substr(4,2))-1,parseInt(this.uploadDateString.substr(6,8))) : null;
        image.type = this.type;
        image.location=this.location;
        image.user=  this.user ? Object.assign(new UserDto(), this.user).ToModel() : null;
        image.url= this.url;
        return image;
    }

    FromModel( image :ImageModel): ImageDto{
      const imageDto = new ImageDto(
      );
      imageDto.id = image.id;
      imageDto.age=image.age;
      imageDto.name=image.name;
      imageDto.uploadDateString = image.uploadDate?  [(image.uploadDate.getFullYear()).toString(), ((image.uploadDate.getMonth()+1)>9 ? '' : '0'),(image.uploadDate.getMonth()+1).toString(), ((image.uploadDate.getDate())>9 ? '' : '0') ,(image.uploadDate.getDate()).toString()].join(''): null;
      imageDto.type = image.type;
      imageDto.location=image.location;
      imageDto.user=  new UserDto().FromModel(image.user);
      imageDto.url= image.url;
      return imageDto;
  }


}
